docs/source/distributed.rst

```diff
@@ -0,0 +1,104 @@
+.. distributed:: torch.
+
+Modes
+=====
+
+There are two modes to control the way calculations are distributed: master-worker mode and process-group mode.
+Each of them is designed to resemble interfaces and APIs familiar to PyTorch users.
+
+The process-group mode API is made to look like the MPI API.
+It is based on the idea that every machine in the network should queue jobs for itself.
+Its interface is designed to give the programmer more flexibility than the master-worker mode which naturally makes its API a little bit more complex.
+
+.. TODO: code examples
+
+The master-worker mode on the other hand is dedicated to the users familiar with Nvidia CUDA.
+The API is simple and makes it easy to start using it right away.
+This model does not scale well due to the bottleneck in the master node which is responsible for all the planning job queuing in all the worker nodes.
+Nevertheless, it is sufficient for networks of few machines.
+
+.. TODO: code examples
+
+Distributed computing
+=====================
+
+In order to configure connections for a program using THD it is required to set a couple of environmental variables.
+Depending on the data channel implementation and selected mode it might be required to set a different subset of those variables.
+
+Environmental variables
+^^^^^^^^^^^^^^^^^^^^^^^
+.. TODO: Explain the usage of the environmental variables.
+
+Here is a list of the environmental variables which configure THD:
+
+- `RANK`
+  It is used to differentiate nodes within the network from each other.
+  .. TODO: Where is it used?
+
+- `WORLD_SIZE`
+  It is used to pass the information about the size of a network to nodes.
+  It might be useful to the master node to know the size of the network to accept connections from all the workers.
+  .. TODO: Where is it used?
+
+- `MASTER_PORT`
+  This variable should be set to the number of a port which worker nodes use to establish connection with a master node.
+
+- `MASTER_ADDR`
+  This variable should set to the address of a master node and the value of `MASTER_PORT` separated by a colon.
+  For example `server15:29500` or `192.168.70.15:29500`.
+
+How to launch distributed computing?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Let's assume we have got a network of 3 machines called node0, node1 and node2.
+Our PyTorch script is located in `~jim/project` and is called `training.py`.
+The examples were tested on the MPICH implementation of the MPI.
```

Comment:
it is pretty relevant as we use options which are specific to MPICH.

See:
- http://stackoverflow.com/a/25289126/4694621
- http://mpich.org/static/docs/v3.2/www1/mpiexec.html

I should have described it better though.